src/tempfile.rs

```diff
@@ -0,0 +1,167 @@
+// Copyright 2017 The Chromium OS Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE-BSD-3-Clause file.
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+//! Create temporary files.
+//!
+//! The temporary files will be created with a name which is available as well
+//! as having an exposed `fs::File` for reading/writing.
+//!
+//! The file will be removed when the object goes out of scope.
+//!
+//! # Examples
+//!
+//! ```
+//! use std::io::Write;
+//! use std::path::Path;
+//! use std::path::PathBuf;
+//! use vmm_sys_util::tempfile::TempFile;
+//! let t = TempFile::new("/tmp/testfile").unwrap();
+//! let mut f = t.as_file();
+//! f.write_all(b"hello world").unwrap();
+//! f.sync_all().unwrap();
+
+use std::ffi::{CString, OsStr};
+use std::fs;
+use std::fs::File;
+use std::os::unix::ffi::{OsStrExt, OsStringExt};
+use std::os::unix::io::FromRawFd;
+use std::path::{Path, PathBuf};
+
+use libc;
+
+use crate::errno::{errno_result, Result, Error};
+
+/// Create and remove a temporary file.  The file will be maintained for the
```

Comment:
The documentation should be split into summary and description. For the summary my suggestion is something like "Wrapper for working with temporary files." or something similar. The summary and description are separated by a blank line and the summary should fit on one line.